gitlab/data_source_gitlab_users.go

```diff
@@ -11,25 +16,62 @@ func dataSourceGitlabUsers() *schema.Resource {
 		Read: dataSourceGitlabUsersRead,
 
 		Schema: map[string]*schema.Schema{
-			"order_by": {
-				Type:     schema.TypeString,
+			"options": {
+				Type:     schema.TypeList,
 				Optional: true,
-				Default:  "id",
-				ValidateFunc: validation.StringInSlice([]string{"id", "name",
-					"username", "created_at"}, true),
-			},
-			"sort": {
-				Type:         schema.TypeString,
-				Optional:     true,
-				Default:      "desc",
-				ValidateFunc: validation.StringInSlice([]string{"desc", "asc"}, true),
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"order_by": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "id",
+							ValidateFunc: validation.StringInSlice([]string{"id", "name",
+								"username", "created_at"}, true),
+						},
+						"sort": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Default:      "desc",
+							ValidateFunc: validation.StringInSlice([]string{"desc", "asc"}, true),
+						},
+						"search": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"active": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"blocked": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"extern_uid": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"provider": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"created_before": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"created_after": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
 			},
 			"users": {
 				Type:     schema.TypeList,
 				Computed: true,
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
-						"id": {
+						"user_id": {
```

Comment:
Why not keep the `id` name to stay consistent with the gitlab API and the sort option specified at line 28?

Reply:
It was to match the other data source gitlab_user.

Reply:
You mean this one https://github.com/claranet/terraform-provider-gitlab/pull/4/files#diff-e5c640cec25ca3e10eb4afecef78833dR15

As both are new, I'd rather use `id` in both